applications/FluidDynamicsApplication/tests/test_apply_mach_dependent_boundary_conditions.py

```diff
@@ -0,0 +1,514 @@
+        outlet_mpart.CreateNewCondition("LineCondition2D2N", 6, [8, 7], props)
+
+        main_mpart.CreateNewCondition("LineCondition2D2N", 7, [7, 4], props)
+        main_mpart.CreateNewCondition("LineCondition2D2N", 8, [4, 1], props)
+
+        for node in main_mpart.Nodes:
+            node.AddDof(KratosMultiphysics.TEMPERATURE)
+            node.AddDof(KratosMultiphysics.DENSITY)
+
+            node.AddDof(KratosMultiphysics.VELOCITY_X)
+            node.AddDof(KratosMultiphysics.VELOCITY_Y)
+            node.AddDof(KratosMultiphysics.VELOCITY_Z)
+
+        self._ResetVariablesToZero()
+
+    def _ResetVariablesToZero(self):
+        for node in self.model["main_model_part"].Nodes:
+            node.SetSolutionStepValue(KratosMultiphysics.TEMPERATURE, 0.0)
+            node.SetSolutionStepValue(KratosMultiphysics.DENSITY, 0.0)
+
+            node.SetSolutionStepValue(KratosMultiphysics.VELOCITY_X, 0.0)
+            node.SetSolutionStepValue(KratosMultiphysics.VELOCITY_Y, 0.0)
+            node.SetSolutionStepValue(KratosMultiphysics.VELOCITY_Z, 0.0)
+
+            node.SetSolutionStepValue(KratosMultiphysics.MOMENTUM_X, 0.0)
+            node.SetSolutionStepValue(KratosMultiphysics.MOMENTUM_Y, 1.0)
+            node.SetSolutionStepValue(KratosMultiphysics.MOMENTUM_Z, 0.0)
+
+    def testDoubleVariable(self):
+        """Tests that the process works well with variables of type array_1d<double, 3> works well with variables of type double."""
+        settings = KratosMultiphysics.Parameters("""
+        {
+            "Parameters" : {
+                "model_part_name" : "main_model_part",
+                "flow_direction_variable" : "MOMENTUM",
+                "subsonic_boundary_conditions" : [
+                    {
+                        "variable_name" : "DENSITY",
+                        "value" : 1.225,
+                        "interval" : [0, "End"]
+                    }
+                ],
+                "supersonic_boundary_conditions" : [
+                    {
+                        "variable_name" : "TEMPERATURE",
+                        "value" : 273.15,
+                        "interval" : [0, "End"]
+                    }
+                ]
+            }
+        }
+        """)
+
+        process = apply_mach_depenedent_boundary_conditions.Factory(settings, self.model)
+        main_model_part = self.model["main_model_part"]
+
+        process.ExecuteInitialize()
+        process.ExecuteBeforeSolutionLoop()
+
+        main_model_part.ProcessInfo.SetValue(KratosMultiphysics.TIME, 1.0)
+
+        # Subsonic case
+        for node in main_model_part.Nodes:
+            node.SetValue(FluidDynamicsApplication.MACH, 0.0)
+
+        process.ExecuteInitializeSolutionStep()
+
+        for node in main_model_part.Nodes:
+            self.assertTrue(node.IsFixed(KratosMultiphysics.DENSITY),
+                msg="Failed to fix subsonic boundary condition (Node #%d)." % node.Id)
```

Comment:
```suggestion
                msg="Failed to fix subsonic boundary condition (Node #{}).".format(node.Id))
```
We normally use `.format` rather than '%'.